src/client/mod.rs

```diff
@@ -979,164 +1229,176 @@ mod test {
         assert_eq!(
             &get_client_value(client)[..],
             "PRIVMSG #test :Hi there!\r\n"
         );
+        Ok(())
     }
 
-    #[test]
-    fn send_no_newline_injection() {
-        let client = IrcClient::from_config(test_config()).unwrap();
-        assert!(
-            client
-                .send(PRIVMSG(format!("#test"), format!("Hi there!\r\nJOIN #bad")))
-                .is_ok()
+    #[tokio::test]
+    async fn send_no_newline_injection() -> Result<(), failure::Error> {
+        let mut client = Client::from_config(test_config()).await?;
+        assert!(client
+            .send(PRIVMSG(format!("#test"), format!("Hi there!\r\nJOIN #bad")))
+            .is_ok());
+        client.stream()?.collect().await?;
+        assert_eq!(
+            &get_client_value(client)[..],
+            "PRIVMSG #test :Hi there!\r\n"
         );
-        assert_eq!(&get_client_value(client)[..], "PRIVMSG #test :Hi there!\r\n");
+        Ok(())
     }
 
-    #[test]
-    fn send_raw_is_really_raw() {
-        let client = IrcClient::from_config(test_config()).unwrap();
-        assert!(
-            client.send(Raw("PASS".to_owned(), vec!["password".to_owned()], None)).is_ok()
-        );
-        assert!(
-            client.send(Raw("NICK".to_owned(), vec!["test".to_owned()], None)).is_ok()
+    #[tokio::test]
+    async fn send_raw_is_really_raw() -> Result<(), failure::Error> {
+        let mut client = Client::from_config(test_config()).await?;
+        assert!(client
+            .send(Raw("PASS".to_owned(), vec!["password".to_owned()], None))
+            .is_ok());
+        assert!(client
+            .send(Raw("NICK".to_owned(), vec!["test".to_owned()], None))
+            .is_ok());
+        client.stream()?.collect().await?;
+        assert_eq!(
+            &get_client_value(client)[..],
+            "PASS password\r\nNICK test\r\n"
         );
-        assert_eq!(&get_client_value(client)[..], "PASS password\r\nNICK test\r\n");
+        Ok(())
     }
 
-    #[test]
+    #[tokio::test]
     #[cfg(not(feature = "nochanlists"))]
-    fn channel_tracking_names() {
+    async fn channel_tracking_names() -> Result<(), failure::Error> {
         let value = ":irc.test.net 353 test = #test :test ~owner &admin\r\n";
-        let client = IrcClient::from_config(Config {
+        let mut client = Client::from_config(Config {
             mock_initial_value: Some(value.to_owned()),
             ..test_config()
-        }).unwrap();
-        client.for_each_incoming(|message| {
-            println!("{:?}", message);
-        }).unwrap();
-        assert_eq!(client.list_channels().unwrap(), vec!["#test".to_owned()])
+        })
+        .await?;
+        client.stream()?.collect().await?;
+        assert_eq!(client.list_channels().unwrap(), vec!["#test".to_owned()]);
+        Ok(())
     }
 
-    #[test]
+    // TODO: this seems to require a somewhat more complex interaction with the
+    // client.
```

Comment:
I don’t follow. This is the same as the next test except that it involves sending a PART.

Reply:
I'll have to comment it out and test it. I don't remember why it didn't work. Will have a look when I port it to stable tokio.